### ISSUE TITLE
Add and populate `Package.platformCompatibility`

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#1429]
As discussed [here](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/discussions/1403#discussioncomment-1766129), add a new field to `Package` where we roll up the platform compatibility from the build matrix.

**Comment by finestructure** [comment]
Updating `Package.platformCompatibility` when we receive build reports will take care of keeping the field up-to-date going forward. Letting the service gradually update via incoming builds will not fill gaps for projects that aren't actively maintained - we need to backfill existing packages.

We have the following options:

1. run the some update sql as a one-off
2. backfill as part of the migration

Option 1 isn't great, because running ad hoc SQL isn't great. The good thing is the query can be run in small badges to limit impact and we can obviously test it on staging. I've run it locally on a copy and it should be fine.

Option 2 would be nice to tie it into the rollout but this would be a quite long-running migration (some minutes at least). Would be tricky to re-run and we obviously can't batch it.

I'd say we merge the PR with the schema change and the continuous update mechanism first and then run the manual SQL on staging for testing.

Before we deploy any of this to production we should make sure the search extension work as expected. That means between the first part of this landing in main and completing the whole task we shouldn't make any deployments. (Or maybe we should deploy from a branch on staging while we integrate everything - but that's a detail.)

The update SQL is as follows:

```sql
UPDATE packages p SET platform_compatibility = ARRAY(
	SELECT
		CASE
			WHEN b.platform LIKE 'macos-%' THEN 'macos'
			ELSE b.platform
		END
	FROM versions v
	JOIN builds b ON b.version_id = v.id
	WHERE v.package_id = p.id
	AND v.latest IS NOT NULL
	AND b.status = 'ok'
	GROUP BY b.platform
	HAVING count(*) > 0
),
updated_at = NOW()
WHERE p.id IN (
	SELECT id FROM packages
        -- adjust the following parameters as needed
	WHERE updated_at < '2021-12-08 12:27:05.82166+00'
	LIMIT 1
);
```

**Comment by finestructure** [comment]
The query syntax for search using the new column is as follows:

```sql
select * from packages where platform_compatibility @> '{linux,macos}';
```

With the platform array it will be easy to support search for single or even multiple platforms, i.e. `platform:linux`, `platform:linux,macos`.

We have 1748 packages supporting Linux - 40%! - which is more than I would have guessed (and that's not counting some that are technically compatible but don't build due to missing OS level dependencies).

**Comment by finestructure** [comment]
Implementation merged here: #1432 

To do:

- [ ] run update SQL on dev
- [ ] deploy to prod
- [ ] run update SQL on prod
